quinn-proto/src/crypto/rustls.rs

```diff
@@ -386,10 +395,15 @@ impl std::error::Error for NoInitialCipherSuite {}
 
 /// A QUIC-compatible TLS server configuration
 ///
-/// Can be constructed via [`ServerConfig::with_single_cert()`][single] or by using the
-/// [`TryFrom`] implementation with a custom [`rustls::ServerConfig`]. A pre-existing
-/// `ServerConfig` must have TLS 1.3 support enabled for this to work. 0-RTT support is
-/// available to clients if `max_early_data_size` is set to `u32::MAX`.
+/// Quinn implicitly constructs a `QuicServerConfig` with reasonable defaults within
+/// [`ServerConfig::with_single_cert()`][single]. Alternatively, `QuicServerConfig`'s [`TryFrom`]
+/// implementation or `with_initial` method can be used to wrap around a custom
+/// [`rustls::ServerConfig`], in which case care should be taken around certain points:
+///
+/// - If `max_early_data_size` is not set to `u32::MAX`, the server may "accept" 0-RTT conversion
+///   attempts by clients but have all attempts by the clients to send 0-RTT data blocked until the
+///   handshake completes. QUIC prohibits `max_early_data_size` values other than 0 or `u32::MAX`.
```

Comment:
I don't think this is correct. If `max_early_data_size` is set to zero, clients may not attempt 0-RTT in future connection attempts at all. See [RFC 9001 § Enabling 0-RTT](https://www.rfc-editor.org/rfc/rfc9001.html#name-enabling-0-rtt). If a server configuration change disables 0-RTT after a client has already cached keys, then 0-RTT data sent with a future connection attempt will be actively rejected (see `WriteError::ZeroRttRejected`), not merely blocked until the handshake completes. This is not to be confused with flow control limits, which may cause 0-RTT to be available but block on use as described here, but this is no different than the 1-RTT behavior.

Reply:
I thought I observed that behavior, but I can no longer reproduce it. Either my initial observation was mistaken or it's been fixed in recent updates.

Reply:
Looks like this hasn't been addressed?

Reply:
Sorry, I confused myself! This should be fixed, and also I made a couple other tweaks to the rustls config docs to remove lingering uses of the word "conversion" as a description of what `into_0rtt` does, to be consistent with your other request.